src/structure.js

```diff
@@ -81,6 +82,20 @@ function Structure (options) {
 inherits(Structure, EventEmitter);
 module.exports = Structure;
 
+Structure.prototype._subscribe = function(path, fn) {
```

Comment:
I'd rather prefer to stick the the immutable style and move the reference "mutation" to outside the functions instead (having the helper functions "pure"). The usage should rather update the reference:

```js
// inside some function with access to "this"
this._referenceListeners = subscribe(this._referenceListeners, somePath, listenerFn);
```